catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -1,39 +1,157 @@
 import * as dateFns from 'date-fns'
+import * as FF from 'final-form'
+import * as FP from 'fp-ts'
+import * as IO from 'io-ts'
 import * as R from 'ramda'
 import * as React from 'react'
-import * as redux from 'react-redux'
-import { Redirect, useHistory } from 'react-router-dom'
-import * as RF from 'redux-form/es/immutable'
+import * as RF from 'react-final-form'
+import * as RRDom from 'react-router-dom'
+import * as urql from 'urql'
 import * as M from '@material-ui/core'
 
 import * as Pagination from 'components/Pagination'
-import * as AuthSelectors from 'containers/Auth/selectors'
 import * as Notifications from 'containers/Notifications'
+import * as Model from 'model'
 import * as APIConnector from 'utils/APIConnector'
-import * as BucketConfig from 'utils/BucketConfig'
-import * as Config from 'utils/Config'
 import Delay from 'utils/Delay'
 import * as Dialogs from 'utils/Dialogs'
 import MetaTitle from 'utils/MetaTitle'
 import * as NamedRoutes from 'utils/NamedRoutes'
 import parseSearch from 'utils/parseSearch'
-import * as Cache from 'utils/ResourceCache'
 import { useTracker } from 'utils/tracking'
+import * as Types from 'utils/types'
 import * as validators from 'utils/validators'
 
 import * as Form from './Form'
 import * as Table from './Table'
-import * as data from './data'
 
-// default icon as returned by the registry
-const DEFAULT_ICON = 'https://d1zvn9rasera71.cloudfront.net/q-128-square.png'
+import BUCKET_CONFIGS_QUERY from './BucketConfigs.generated'
+import ADD_MUTATION from './BucketsAdd.generated'
+import UPDATE_MUTATION from './BucketsUpdate.generated'
+import REMOVE_MUTATION from './BucketsRemove.generated'
+import { BucketConfigSelectionFragment as BucketConfig } from './BucketConfigSelection.generated'
+
+// TODO: move into some shared utility module
+function assertNever(x: never): never {
+  throw new Error(`Unexpected value '${x}'. Should have been never.`)
+}
+
+const SNS_ARN_RE = /^arn:aws(-|\w)*:sns:(-|\w)*:\d*:\S+$/
 
 const DO_NOT_SUBSCRIBE_STR = 'DO_NOT_SUBSCRIBE'
 const DO_NOT_SUBSCRIBE_SYM = Symbol(DO_NOT_SUBSCRIBE_STR)
 
-const SNS_ARN_RE = /^arn:aws(-|\w)*:sns:(-|\w)*:\d*:\S+$/
+type SnsFormValue = string | typeof DO_NOT_SUBSCRIBE_SYM
+// eslint-disable-next-line @typescript-eslint/no-redeclare
```

Comment:
it's a little tricky when you want to declare some kind of [type representative](https://github.com/fantasyland/fantasy-land#type-representatives) (couldnt find a better explanation for this term). like, when you want to define a `io-ts` codec with the same name as the type it represents for convenience, to avoid extra names like `SmthCodec` or `SmthValue` and such.
also, i see eslint rules as something that prevents us from doing some stupid things by accident, not as dogmas we must obey even if they stand in our way.